### PR TITLE
Avg docs

### PR DESCRIPTION
**Pull request by arigdon-usgs**
This is a work in progress. Created a python notebook to be used for comparisons. Also cleaned up AverageAlgorithm.py. The metadata was being set twice so I removed some of the redundancy. Also observatories had to be set explicitly but that was changed so if observatories were not set it defaults to the observatories in the stream. Also added better compatibility with setting the observatory scale factors.

### CODE REVIEW

**Comment by jmfee-usgs** [comment]
This is a review in process (jk).  I took a look and I think you're making progress.

- should the "-checkpoint.ipynb" files be included, or added to `.gitignore`?


- This could be more pythonic:
https://github.com/usgs/geomag-algorithms/pull/215/files#diff-af6ea390026fd7000bd3913c33546396R103
```
self.observatories = self.observatories or [t.stats.station for trace in timeseries]
```

- do not use the global `CORR`.

There may be multiple instances of the algorithm that require different correction factors; so it should be an instance variable and not a global.

You _could_ use it for defaults if no scales are specified, but I think the default for averaging should be unscaled.

Maybe make the algorithm using a dictionary of scale factors instead of an array, and populate that dictionary during configure instead of during process...  If called via controller it will likely have observatories and scales.  If called another way, the scales can be manually configured or use the default (unscaled).


- default scales to `None`, and remove this section entirely (see above and below).

https://github.com/usgs/geomag-algorithms/pull/215/files#diff-af6ea390026fd7000bd3913c33546396R109


- Only set default once (and don't set if not needed)

You already set a default here:
    https://github.com/usgs/geomag-algorithms/pull/215/files#diff-af6ea390026fd7000bd3913c33546396R126

So this seems unnecessary:
    https://github.com/usgs/geomag-algorithms/pull/215/files#diff-af6ea390026fd7000bd3913c33546396R111


- watch variable naming conventions:

https://github.com/usgs/geomag-algorithms/pull/215/files#diff-af6ea390026fd7000bd3913c33546396R204


- move get_trace to TimeseriesUtility

**Comment by erigler-usgs** [comment]
This PR was replace by #217. Closing.

File: geomagio/algorithm/AverageAlgorithm.py [hunk 1]
```diff
@@ -93,26 +82,20 @@ def process(self, timeseries):
         out_stream:
             new stream object containing the averaged values.
         """
+        if not self.observatories:
```
[the remark this answers is not in the document]
Comment (erigler-usgs) [reply geomagio/algorithm/AverageAlgorithm.py]: I fixed this, as well as the default assignments that follow.

File: geomagio/algorithm/AverageAlgorithm.py [hunk 2]
```diff
@@ -93,26 +82,20 @@ def process(self, timeseries):
...
-        # Set Correction values if specified and add a dicitonary
-        # if observatory is not already set in CORR
-        if self.scales[0]:
-            for obs in self.observatories:
-                if obs not in CORR:
-                    new_obs = {str(obs): 1.0}
-                    CORR.update(new_obs)
-            for (i, obs) in enumerate(self.observatories):
-                CORR[obs] = self.scales[i]
+        scale_values = self.scales or ([1] * len(timeseries))
```
[the remark this answers is not in the document]
Comment (erigler-usgs) [reply geomagio/algorithm/AverageAlgorithm.py]: I will stick with Adren's original logic for now. There is very little risk of multiple traces from the same observatory and same channel, and I am not comfortable enough with Adren's overall logic, or your proposal, to make this change myself at this time.

File: geomagio/algorithm/AverageAlgorithm.py [hunk 3]
```diff
@@ -136,8 +118,16 @@ def process(self, timeseries):
         dst_tot = numpy.mean(combined, axis=0)
 
         # Create a stream from the trace function
+        new_stats = obspy.core.Stats()
+        new_stats.station = 'USGS'
+        new_stats.channel = self.outchannel
+        new_stats.network = 'NT'
+        new_stats.location = 'RD'
```
[the remark this answers is not in the document]
Comment (erigler-usgs) [reply geomagio/algorithm/AverageAlgorithm.py]: I think I figured this out, but please check my logic in the new PR.

File: geomagio/algorithm/AverageAlgorithm.py [hunk 4]
```diff
@@ -152,7 +142,7 @@ def add_arguments(cls, parser):
             command line argument parser
         """
         parser.add_argument('--average-observatory-scale',
-               default=(None,),
+               default=None,
```
[the remark this answers is not in the document]
Comment (erigler-usgs) [reply geomagio/algorithm/AverageAlgorithm.py]: I would rather stick with Adren's original logic here, where the scale values are specified separately from the observatories. I can see the benefit of your proposal, but I'd rather not break something that works for the moment.

**Pull request closed without merging**